code/game/turfs/simulated/walls.dm

```diff
@@ -82,6 +84,13 @@
 	return "You can deconstruct this by welding it, and then wrenching the girder.<br>\
 			You can build a wall by using metal sheets and making a girder, then adding more metal or plasteel."
 
+/turf/simulated/wall/proc/rust()
+	if(rusted)
+		return
+	name = "rusted [name]"
+	add_overlay(icon('icons/turf/overlays.dmi', pick("rust", "rust2"), pick(NORTH, SOUTH, EAST, WEST)))
```

Comment:
can you move this part to `update_icon()`?
That way there's a unified place for handling all walls overlays, in line with the upcoming appearance refactor.
Also gives a way to un-rust walls if need be.
Also it becomes much easier to (un)rust walls via VV or similar.